src/use_case/GeneratePassword/GeneratePasswordInteractor.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    /**
+     * Method which contains the logic for the Generate Password use case (interactor) which is triggered to complete use case
+     *
+     * @param generatePasswordInputData The input data for the use case interactor
+     */
+    @Override
+    public void execute(GeneratePasswordInputData generatePasswordInputData) {
+        if (!generatePasswordInputData.isLowerAlpha() && !generatePasswordInputData.isUpperAlpha() &&
+                !generatePasswordInputData.isNumericalChars() && !generatePasswordInputData.isExtendedAscii() &&
+                !generatePasswordInputData.isPunctuationOne() && !generatePasswordInputData.isPunctuationTwo() &&
+                !generatePasswordInputData.isPunctuationThree() && !generatePasswordInputData.isPunctuationFour() &&
+                !generatePasswordInputData.isPunctuationFive() &&
+                        generatePasswordInputData.getAlsoIncludeFrom().equals("") &&
+                        generatePasswordInputData.getExcludeFrom().equals("")) {
+            generatePasswordPresenter.prepareFailView("You have not selected any characters for your character set.");
+        }
+        else {
+            String generatedPassword = generatePassword(generatePasswordInputData.getPasswordLength(),
+                    generatePasswordInputData.isLowerAlpha(), generatePasswordInputData.isUpperAlpha(),
+                    generatePasswordInputData.isNumericalChars(), generatePasswordInputData.isExtendedAscii(),
+                    generatePasswordInputData.isPunctuationOne(), generatePasswordInputData.isPunctuationTwo(),
+                    generatePasswordInputData.isPunctuationThree(), generatePasswordInputData.isPunctuationFour(),
+                    generatePasswordInputData.isPunctuationFive(), generatePasswordInputData.getAlsoIncludeFrom(),
+                    generatePasswordInputData.getExcludeFrom());
+            System.out.println(generatedPassword);
+            GeneratePasswordOutputData generatePasswordOutputData = new GeneratePasswordOutputData(false, generatedPassword);
+            generatePasswordPresenter.prepareSuccessView(generatePasswordOutputData);
+        }
+    }
+
+    /**
+     * Method which generates and returns a password given certain charset specifications
+     * @param length The password length
+     * @param lowerAlpha Lowercase charset
+     * @param upperAlpha Uppercase charset
+     * @param numericalChars Numerical charset
+     * @param extendedAscii Extended ASCII charset
+     * @param punctuationOne PunctuationOne charset
+     * @param punctuationTwo PunctuationTwo charset
+     * @param punctuationThree PunctuationThree charset
+     * @param punctuationFour PunctuationFour charset
+     * @param punctuationFive PunctuationFive charset
+     * @param alsoIncludeFrom Extra charset inclusion
+     * @param excludeFrom Charset exlusions
+     * @return A randomly generated password of a given length using the specified charsets
+     */
```

Comment:
Good idea using a helper method for the password generator, seems like a lot of lines of code!